### ISSUE TITLE
Package cannot be resolved with `--verifytypes`

### ISSUE THREAD

**Issue by vberlier** [microsoft/pyright#2320]
With version `1.1.170` the `--verifytypes` option returns an error:

```
Package name: "tokenstream"
Package directory: ""
  error: Package "tokenstream" cannot be resolved
...
```

This used to work fine in `1.1.169`:

```
Package name: "tokenstream"
Package directory: "tokenstream"
Path of py.typed file: "tokenstream/py.typed"
```

full log `1.1.169` https://github.com/vberlier/tokenstream/runs/3621648131#step:10:10
full log `1.1.170` https://github.com/vberlier/tokenstream/runs/3663617360#step:10:10

**Comment by erictraut** [comment]
I'm excited to hear that you're using `--verifytypes`. 

Let's try to figure out what's causing the apparent regression.

I tried to repro by installing the latest version of tokenstream locally using the commands:
```
pip install git+https://github.com/vberlier/tokenstream
pyright --verifytypes tokenstream
```

This didn't produce any errors.

Are you sure that your CI script is actually installing the library? The `--verifytypes` command assumes that the library has been installed in the local Python environment.

**Comment by jakebailey** [comment]
I've also tried this locally (on Linux), doing the same thing that CI task does and can't reproduce it. This must be something special about this CI environment.

I have one thought as to what this may be; I recently did a refactor to explicitly handle empty execution environments; the type verifier is one of those cases that uses an unrooted execution environment, as it's only intended to be run on installed packages, not local folders. Previously, it was a bit unclear what our behavior would be with that spurious empty path; it could very well be that this only worked because the empty root path was somehow interpreted as the working directory when in this mode in the right conditions. For this case, that would mean that the local copy of `tokenstream` was found as it's in the repo root as the CI's cwd.

But, that doesn't explain why it works at all locally.

**Comment by erictraut** [comment]
@vberlier, any other thoughts on this issue? The fact that it works locally seems to point to some configuration issue with CI, so I don't know that there's anything to fix in pyright. My inclination is to close the bug report. Let us know if there's anything else we can do to help you diagnose the CI issue.

**Comment by vberlier** [comment]
I investigated a bit and maybe it's poetry that doesn't fully install the package as pyright seems to expect. When I checkout tokenstream and poetry install and then run the latest pyright version I get the same error message as in github actions. I also tried to find other projects using both poetry and pyright with the `--verifytypes` option but they don't do anything differently. Though the projects I found weren't using the latest version of pyright so maybe they'll run into the same problem too when they update. Anyway I'm gonna keep looking into it, thanks for your help.

**Comment by AbstractUmbra** [comment]
Sorry to bump a closed issue, but I am also suffering this problem, both locally and in gh actions. I suspect this may be more of a `poetry` problem than `pyright` but wanted to ask if there was anything more here?

For more context, I am within the correct venv (which is not installed in the project) and this is my output:
```sh
(hondana-epKOD00U-py3.9)  π hondana main ✗ ❯ poetry run pyright --ignoreexternal --lib --verifytypes hondana  
Package name: "hondana"
Package directory: ""
  error: Package "hondana" cannot be resolved

Symbols used in public interface:

Symbols exported by "hondana": 0
  With known type: 0
  With partially unknown type: 0
    (Ignoring unknown types imported from other packages)
  Functions without docstring: 0
  Functions without default param: 0
  Classes without docstring: 0

Other symbols referenced but not exported by "hondana": 0
  With known type: 0
  With partially unknown type: 0

Type completeness score: 0%

Completed in 0.097sec
```

I have tried with `poetry run pyright` and `pyright` and both have the same outcome.

**Comment by vberlier** [comment]
@AbstractUmbra I finally figured out what was going on. Basically the `--verifytypes` option gets confused if there's a directory with the same name (your source directory) in the current directory. It's kind of a hack but this works:

```yml
      - name: Verify library interface
        run: mkdir _tmp && cd _tmp && poetry run npx pyright --verifytypes tokenstream
```

**Comment by AbstractUmbra** [comment]
> @AbstractUmbra I finally figured out what was going on. Basically the `--verifytypes` option gets confused if there's a directory with the same name (your source directory) in the current directory. It's kind of a hack but this works:
> 
> ```yaml
>       - name: Verify library interface
>         run: mkdir _tmp && cd _tmp && poetry run npx pyright --verifytypes tokenstream
> ```

I appreciate this greatly, thank you!
I'm wondering though - is this perhaps a bug with pyright? The behaviour clearly changed.

**Comment by erictraut** [comment]
@AbstractUmbra, if you think this might be a bug, please open a new issue with repro details.

**Comment by tommyjcarpenter** [comment]
I am having this same issue as @vberlier commented
